### PR TITLE
[DPE-4236] Nodeport upgrade

### PR DESCRIPTION
**Pull request by dragomirp**
The service gets overridden during upgrade and the node port configuration gets lost.

Changed to create a separate service for nodeport.

### CODE REVIEW

File: src/charm.py
Comment (dragomirp) [inline src/charm.py:1]: Creating a separate service for the node port. That should persist between upgrades.

File: src/relations/backend_database.py
Comment (dragomirp) [inline src/relations/backend_database.py:1]: Saw those fail on model removal

File: tests/integration/relations/pgbouncer_provider/helpers.py
Comment (dragomirp) [inline tests/integration/relations/pgbouncer_provider/helpers.py:1]: Copypaste from the VM charm

File: tests/integration/test_node_port.py [hunk 1]
```diff
@@ -147,23 +148,9 @@ async def test_node_port_and_clusterip_setup(ops_test: OpsTest):
...
-    userpass = await get_juju_secret(ops_test, info["secret-user"])
-    host, nodeport = info["endpoints"].split(":")
-
-    connection_string = (
-        f"dbname='{info['database']}' user='{userpass['username']}'"
-        f" host='{host}' port='{nodeport}' "
-        f"password='{userpass['password']}' connect_timeout=10"
+    credentials = await fetch_action_get_credentials(
+        ops_test.model.applications[DATA_INTEGRATOR].units[0]
     )
```
Comment (dragomirp) [inline tests/integration/test_node_port.py:155]: Getting the creds from data-integrator should work on either Juju

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -470,14 +471,54 @@ def test_raise_untrusted_error(self, _, __, _deployed_without_trust):
         self.charm._node_name
         self.charm._node_ip
         self.charm._node_port("port")
-        self.charm.patch_port()
```
Comment (dragomirp) [inline tests/unit/test_charm.py:473]: Stopped calling get

File: src/charm.py [hunk 3]
```diff
@@ -257,59 +258,69 @@ def get_node_ports(self) -> Dict[str, Any]:
...
-            client.patch(
-                res=lightkube.resources.core_v1.Service,
-                obj=service,
-                name=service.metadata.name,
-                namespace=service.metadata.namespace,
-                force=True,
-                field_manager=self.app.name,
-                patch_type=lightkube.types.PatchType.MERGE,
-            )
-            logger.debug("Patched k8s service")
```
Comment (dragomirp) [inline src/charm.py:306]: Should we keep on patching the ports on the juju service?

**Review by taurus-forever — APPROVED** [review]
Thank you!

File: src/charm.py (hunk 3 above)
Comment (marceloneppel) [reply src/charm.py:306]: I don't think so. I remember Mykola told me that etcd is sensible to write.

**Review by marceloneppel — APPROVED** [review]
Looks great! Thanks!

**Review by lucasgameiroborges — APPROVED** [review]

**Review by phvalguima — APPROVED** [review]
LGTM

**Pull request merged**